core-rust/state-manager/src/metrics.rs

```diff
@@ -387,3 +403,64 @@ impl MetricLabel for VertexPrepareStopReason {
         }
     }
 }
+
+/// A number of most recent [`RoundSlot`]s of a single validator to track for metrics purposes.
+const PROPOSAL_HISTORY_LEN: usize = 100;
```

Comment:
_(note from author)_

@LukasGasior1 this is the `N` you used - do you care to suggest some actual value :D ?
(mine was not engineered at all)